plugins/woocommerce/client/blocks/assets/js/blocks/add-to-cart-with-options/grouped-product-selector/product-item-cta/frontend.ts

```diff
@@ -2,8 +2,97 @@
  * External dependencies
  */
 import { store } from '@wordpress/interactivity';
+import type { HTMLElementEvent } from '@woocommerce/types';
+
+/**
+ * Internal dependencies
+ */
+import type { AddToCartWithOptionsStore } from '../../frontend';
+
+// Stores are locked to prevent 3PD usage until the API is stable.
+const universalLock =
+	'I acknowledge that using a private store means my plugin will inevitably break on the next store release.';
+
+const getInputElementFromEvent = (
+	event: HTMLElementEvent< HTMLButtonElement >
+) => {
+	const target = event.target as HTMLButtonElement;
+
+	const inputElement = target.parentElement?.querySelector(
+		'.input-text.qty.text'
+	) as HTMLInputElement | null | undefined;
```

Comment:
AFAIK it either returns element or `null`:

```suggestion
	) as HTMLInputElement | null;
```